src/axelrod_dojo/archetypes/fsm.py

```diff
@@ -140,34 +140,30 @@ def parse_repr(cls, s):
         return cls(num_states, rows, initial_state, initial_action)
 
     def receive_vector(self, vector):
-        """Receives a vector and creates an instance attribute called
-        vector."""
-        self.vector = vector
-
-    def vector_to_instance(self):
-        """Turns the attribute vector in to a FSM player instance.
+        """
+        Read a serialized vector into the set of FSM parameters (less initial
+        state).  Then assign those FSM parameters to this class instance.
 
         The vector has three parts. The first is used to define the next state 
         (for each of the player's states - for each opponents action).
 
         The second part is the player's next moves (for each state - for 
         each opponent's actions).
 
-        Finally, a probability to determine the player's first move."""
-
-        num_states = int((len(self.vector) - 1) / 4)
-        state_scale = self.vector[:num_states * 2]
-        next_states = [int(s * (num_states - 1)) for s in state_scale]
-        actions = self.vector[num_states * 2: -1]
-        starting_move = C if round(self.vector[-1]) == 0 else D
+        Finally, a probability to determine the player's first move.
+        """
+        state_scale = vector[:self.num_states * 2]
+        next_states = [int(s * (self.num_states - 1)) for s in state_scale]
```

Comment:
same comment on `int`

Reply:
s should be a decimal in fact.

The PSO doesn't totally make for FSM...  This line converts a continuous number into a discrete state...